spec/rubocop/cop/rails/i18n_lazy_lookup_spec.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Rails::I18nLazyLookup do
+  subject(:cop) { described_class.new }
+
+  it 'registers an offense and corrects when using translation helpers with the key scoped to controller and action' do
+    expect_offense(<<~RUBY)
+      class FooController
+        def action
+          t 'foo.action.key'
+            ^^^^^^^^^^^^^^^^ Use "lazy" lookup for the text used in controllers.
+          translate 'foo.action.key'
+                    ^^^^^^^^^^^^^^^^ Use "lazy" lookup for the text used in controllers.
+        end
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      class FooController
+        def action
+          t '.key'
+          translate '.key'
+        end
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when translation methods scoped to `I18n`' do
+    expect_no_offenses(<<~RUBY)
+      class FooController
+        def action
+          I18n.t 'foo.action.key'
+          I18n.translate 'foo.action.key'
+        end
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when not inside controller' do
+    expect_no_offenses(<<~RUBY)
+      class FooService
+        def do_something
+          t 'foo_service.do_something.key'
+        end
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when not inside controller action' do
+    expect_no_offenses(<<~RUBY)
+      class FooController
+        private
+
+        def action
+          t 'foo.action.key'
+        end
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when translating key not scoped to controller and action' do
+    expect_no_offenses(<<~RUBY)
+      class FooController
+        def action
+          t 'one.two.key'
+        end
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when using "lazy" translation' do
+    expect_no_offenses(<<~RUBY)
+      class FooController
+        def action
+          t '.key'
+        end
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when translation key is not a string nor a symbol' do
+    expect_no_offenses(<<~RUBY)
+      class FooController
+        def action
+          t ['foo.action.key']
```

Comment:
Is this legal?

Reply:
Rails' `t` helpers delegate to `I18n.translate` and this is used for [bulk translates](https://github.com/ruby-i18n/i18n/blob/dbee3d8e08c63c5a868cb097b16b3a1ac6310ae4/lib/i18n.rb#L151-L160).

Reply:
Tested this and found a couple of notable things:

1. `t ['.key']` won't work as doesn't Rails allow lazy lookup for bulk lookup. So this example in question is perfectly fine.

2. What troubled me is [the implementation of `translate` in controller code](https://github.com/rails/rails/blob/b6f4177a6d6361adcb802d5adfb5e1cb46c359fe/actionpack/lib/abstract_controller/translation.rb#L18):
```ruby
> ['.a', '.b'].start_with?('.')
NoMethodError: undefined method `start_with?' for [".a", ".b"]:Array
```

It would work before Rails 7, because it was implemented as:
```ruby
if key.to_s.first == "."
```
where `key` is `['.a', '.b']`, but I suspect it would fail in Rails 7.

3. `AbstractController::Translation` is not included to [`ActionController::API`](https://github.com/rails/rails/blob/main/actionpack/lib/action_controller/api.rb), only to `ActionController::Base`](https://github.com/rails/rails/blob/main/actionpack/lib/action_controller/base.rb). It doesn't make the cop unsafe, since it's not us who suggest using `t`, it's already there in the code.

Reply:
Thanks for investigation 👍 
Yes, I think no one actually uses this bulk feature, since, as you mentioned, it doesn't work.

This test actually tests that *no* offenses are generated for this case. Only when the keys is a string or a symbol.

So, wdyt, should I remove this test case or left it as is?

Reply:
It's fine to keep it, just to make sure we don't add to the existing mess 😄